frontend/packages/noobaa-storage-plugin/src/components/data-resiliency-card/data-resiliency-card.tsx

```diff
@@ -0,0 +1,130 @@
+import * as React from 'react';
+import { Icon } from 'patternfly-react';
+import { Progress, pluralize } from '@patternfly/react-core';
+
+import './data-resiliency-card.scss';
+
+import { DashboardCard } from '@console/internal/components/dashboard/dashboard-card/card';
+import { DashboardCardBody } from '@console/internal/components/dashboard/dashboard-card/card-body';
+import { DashboardCardHeader } from '@console/internal/components/dashboard/dashboard-card/card-header';
+import { DashboardCardTitle } from '@console/internal/components/dashboard/dashboard-card/card-title';
+import {
+  withDashboardResources,
+  DashboardItemProps,
+} from '@console/internal/components/dashboards-page/with-dashboard-resources';
+import { DashboardCardHelp } from '@console/internal/components/dashboard/dashboard-card';
+
+import { DATA_RESILIENCE_QUERIES } from '../../queries';
+import { getPropsData } from '../../utils';
+
+const getFormattedEta = (eta) => {
+  if (eta < 60) {
+    return pluralize(eta, 'second');
+  }
+  if (eta < 3600) {
+    return pluralize(Math.round(eta / 60), 'minute');
+  }
+  return pluralize(Math.round(eta / 3600), 'hour');
+};
+
+const DataResiliencyStatusBody: React.FC<DataResiliencyStatusBody> = ({ isResilient }) =>
+  isResilient ? (
+    <>
+      <div className="nb-data-resiliency__status-title--ok">Your Data is Resilient</div>
+      <div className="nb-data-resiliency__icon--ok">
+        <Icon type="fa" name="check-circle" size="5x" />
+      </div>
+    </>
+  ) : (
+    <>
+      <div className="nb-data-resiliency__icon--error">
+        <Icon type="fa" name="exclamation-triangle" size="5x" />
+      </div>
+      <div className="nb-data-resiliency__status-title--error">No data available</div>
+    </>
+  );
+
+const DataResiliencyBuildBody: React.FC<DataResiliencyBuildBody> = ({
+  progressPercentage,
+  eta,
+}) => {
+  const formattedEta = getFormattedEta(eta);
+  return (
+    <>
+      <div className="nb-data-resiliency__title">Rebuilding data</div>
+      <Progress
+        className="nb-data-resiliency__utilization-bar"
+        value={progressPercentage}
+        title="Rebuilding in Progress"
+        label={`${progressPercentage}%`}
+      />
+      <span className="text-secondary nb-data-resiliency__eta">
+        Estimating {formattedEta} to completion
```

Comment:
What if the eta is not available?

Reply:
thanx for pointing out !

Reply:
@cloudbehl previously it was supposed to show `0 seconds`(error in that as well) but I have changed it now.
According to my understanding, we should not show eta if it is not able to be fetched from prom.
So, I am not showing it in updated code.
Thoughts ?

Reply:
refer to the screenshots I have updated!